docs/permission/writing-a-policy.md

```diff
@@ -3,3 +3,137 @@ id: writing-a-policy
+
+## Resource types
+
+Now let's say we would also like to restrict users from viewing catalog entities that they do not own, just like we did for unregistering entities. One way to achieve this may be to simply duplicate our if statement and check for the `catalog.entti.read` permission:
+
+```diff
+    async handle(
+      request: PolicyAuthorizeQuery,
+      user?: BackstageIdentityResponse,
+     ): Promise<PolicyDecision> {
+      if (request.permission.name === 'catalog.entity.delete') {
+        return createCatalogPolicyDecision(
+          catalogConditions.isEntityOwner(
+            user?.identity.ownershipEntityRefs ?? [],
+          ),
+        );
+      }
+
++     if (request.permission.name === 'catalog.entity.read') {
++       return createCatalogPolicyDecision(
++         catalogConditions.isEntityOwner(
++           user?.identity.ownershipEntityRefs ?? [],
++         ),
++       );
++     }
+
+      return { result: AuthorizeResult.ALLOW };
+    }
+```
+
+If you choose to write your policy this way, it will certainly work! You should be able to verify this by saving this policy and seeing that the catalog now only shows the entities that you own. However, you can imagine that as policies grow to handle many different permissions, these conditionals can quickly become repetitive. We can author this same policy in a more scalable way by using resource types.
+
+```diff
+    async handle(
+      request: PolicyAuthorizeQuery,
+      user?: BackstageIdentityResponse,
+     ): Promise<PolicyDecision> {
+-     if (request.permission.name === 'catalog.entity.delete') {
++     if (request.permission.resourceType === 'catalog-entity') {
+        return createCatalogPolicyDecision(
+          catalogConditions.isEntityOwner(
+            user?.identity.ownershipEntityRefs ?? [],
+          ),
+        );
+      }
+
+-     if (request.permission.name === 'catalog.entity.read') {
+-       return createCatalogPolicyDecision(
+-         catalogConditions.isEntityOwner(
+-           user?.identity.ownershipEntityRefs ?? [],
+-         ),
+-       );
+-     }
+-
+      return { result: AuthorizeResult.ALLOW };
+    }
+```
+
+In this example, we use the `catalog-entity` resource type to catch all authorization requests that have to do with resources from the catalog. Now, you should be able to see the same functionality as before (only see the catalog entities that you own) - success!
+
+## Conclusion
+
+Through a combination of permissions and conditions, you should be able to author a policy that works for your instance of Backstage and your organzation. As the ecosystem around permissions in Backstage become more mature, you may be able to choose from other authorization packages instead of writing your own policy. If you're interested in more detailed descriptions of the concepts that comprise the permission framework, check out the [Concepts page](./concepts.md).
```

Comment:
```suggestion
As the ecosystem around permissions in Backstage become more mature, you may be able to choose from other authorization packages instead of writing your own policy.
```

Not sure about this one - I want to allude to possible expansions of the framework that will make it more powerful & nicer to use, without straight up just pointing to the commercial packages that are on the roadmap.

Reply:
Yeah, I think this is perfect for now. Obviously it means the docs will have to respond as the ecosystem evolves, but I think that's fine!